### ISSUE TITLE
Investigate solution for managing IP address space in infrastructure / clusters

### ISSUE THREAD

**Issue by jhouse-solvd** [department-of-veterans-affairs/va.gov-team#19579]
## Description
Look into running calico or similar solution so we don't have to worry about IP space in any of our clusters

## Background/context/resources
_Any additional context for the reader to know, if applicable_ 

## Technical notes
_Notes around work that is happening, if applicable_

---
## Tasks
- [x] Experiment with new cluster based on our needs to understand how it all works 

## Definition of Done
- [x] Working solution for the management of IP address space within EKS clusters

---
### Reminders
- [x] Please attach your team label and any other appropriate label(s) 
- [x] Please attach the needs grooming tag if needed
- [x] Please connect to an epic

**Comment by jhouse-solvd** [comment]
from @mleclerc00: 

While calico is really powerful, it looks like something that would take a lot of effort up front and down the line to maintain.  As I posted in the thread last week, Amazon strongly suggests against using a custom CNI.  The other thing to note is the fact that it would break some kubernetes api functionality for the following reasons:

https://medium.com/faun/choosing-your-cni-with-aws-eks-vpc-cni-or-calico-1ee6229297c5

“For every EKS cluster, AWS launches the control plane in a VPC which is managed by AWS, which is also explained in the AWS docs. What the doc doesn’t explain is that at the time of creation, a route table entry for your VPC CIDR to the ENI used to connect to your VPC is also created. This is how the API server is able to communicate to all the Nodes and pods in the cluster, but if we decide to use our own CNI, like calico, we change the POD CIDR, and now the API server has no clue on how to route the traffic to the pod IP. So effectively, as per the above diagram, our POD CIDR is of the range 10.5.0.0/16 which is not routable by the API server due to the missing route table entry in the AWS managed VPC.”

An alternate solution is that we can use secondary CIDRs for our worker instances to use. https://www.eksworkshop.com/beginner/160_advanced-networking/secondary_cidr/

**Comment by tskinn** [comment]
Matt has successfully created an EKS cluster in the vagov-utility cluster that utilizes the calico network CNI and we have successfully deployed a few applications to it. 

We experienced the documented issue of the Kubernetes master nodes not being able to route traffic to pods that need it. The alb controller pod needs to receive webhooks from the Kubernetes masters in order to do its thing. This was fixed by running the alb controller pod in network=host mode. I think finding out which applications will need the same treatment will be pretty easy.
